lib/semantics/resolve-labels.cc

```diff
@@ -898,13 +898,11 @@ void CheckLabelDoConstraints(const SourceStmtList &dos,
               TargetStatementEnum::CompatibleDo)) {
         errorHandler.Say(doTarget.parserCharBlock,
             parser::MessageFormattedText{
-                "Only an END DO or CONTINUE must be used to terminate a labeled DO loop"_err_en_US,
-                SayLabel(label)});
+                "Only an END DO or CONTINUE must be used to terminate a labeled DO loop"_err_en_US});
```

Comment:
That's awkward English usage.  Try "An END DO or CONTINUE statement must..." or "Only an END DO or CONTINUE statement may...".

Reply:
I agree that the English seems awkward.

How about "Either" in both cases: "Either an END DO or CONTINUE must ..." and "Either an END DO or CONTINUE should ..."

Reply:
I think it is good to use similar wording in both cases. Any of these seem fine to me:
"END DO or CONTINUE must/should be used to terminate a labeled DO loop"
"An END DO or CONTINUE must/should be used to terminate a labeled DO loop"
"Either END DO or CONTINUE must/should be used to terminate a labeled DO loop"

Reply:
What we really mean is that the terminal statement of a labeled DO loop must be an END DO or CONTINUE statement.  So try wording it like that, with the offending statement of the error message also being the subject of the sentence that's describing what's wrong with it.

Reply:
Actually, after thinking about @schweitzpgi's comment, I don't think that we want to have different messages for warnings and errors.  Rather, I think that, for the case where the code is both valid and does not conform to the standard, and the -Wstandard option is in effect, we should issue a warning message about using END DO or CONTINUE.  If the -Werror option is in effect, that message should be an error.

I'll create a separate issue for differentiating the case where the DO loop termination is invalid versus not conforming to the standard.